shared/src/commonTest/kotlin/co/nimblehq/avishek/phong/kmmic/domain.usecase/GetSurveysUseCaseTest.kt

```diff
@@ -0,0 +1,68 @@
+package co.nimblehq.avishek.phong.kmmic.domain.usecase
+
+import co.nimblehq.avishek.phong.kmmic.domain.model.Survey
+import co.nimblehq.avishek.phong.kmmic.domain.repository.SurveyRepository
+import io.kotest.matchers.shouldBe
+import io.mockative.*
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+class GetSurveysUseCaseTest {
+
+    @Mock
+    private val mockRepository = mock(classOf<SurveyRepository>())
+
+    private lateinit var useCase: GetSurveysUseCase
+
+    private val mockThrowable = Throwable("mock")
+    private val mockSurvey = Survey(
+        "id",
+        "title",
+        "description",
+        true,
+        "coverImageUrl"
+    )
+
+    @BeforeTest
+    fun setUp() {
+        useCase = GetSurveysUseCaseImpl(mockRepository)
+    }
+
+    @Test
+    fun `when getSurveys is called successfully, it returns a list of surveys`() = runTest {
+        given(mockRepository)
+            .function(mockRepository::getSurveys)
+            .whenInvokedWith(any(), any(), any())
+            .thenReturn(
+                flow {
+                    emit(listOf(mockSurvey))
```

Comment:
Then we can use it here like this.
```suggestion
                    emit(listOf(MockUtil.survey))
```